packages/geo/src/Providers/AmazonLocationServicesProvider.ts

```diff
@@ -102,4 +116,95 @@ export class AmazonLocationServicesProvider implements GeoProvider {
 
 		return { mapName, style };
 	}
+
+	/**
+	 * Search by text input with optional parameters
+	 * @param  {string} text - The text string that is to be searched for
+	 * @param  {SearchByTextOptions} options? - Optional parameters to the search
+	 * @returns {Promise<Place[]>} - Promise resolves to a list of Places that match search parameters
+	 */
+	public async searchByText(
+		text: string,
+		options?: SearchByTextOptions
+	): Promise<Place[]> {
+		try {
+			const credentialsOK = await this._ensureCredentials();
+			if (!credentialsOK) {
+				return Promise.reject('No credentials');
+			}
+		} catch (error) {
+			throw '';
+		}
```

Comment:
I suggest this simpler way to handle this: 
```suggestion
		const credentialsOK = await this._ensureCredentials();

		if (!credentialsOK) {
			throw 'No credentials';
		}
```